pallets/pallet-did-lookup/src/migrations.rs

```diff
@@ -16,28 +16,124 @@
 
 // If you feel like getting in touch with us, you can do so at info@botlabs.org
 
-use crate::{linkable_account::LinkableAccountId, AccountIdOf, Config, ConnectionRecordOf, DidIdentifierOf, Pallet};
-
-use crate::{ConnectedAccounts as ConnectedAccountsV2, ConnectedDids as ConnectedDidsV2};
+use crate::{
+	linkable_account::LinkableAccountId, AccountIdOf, Config, ConnectedAccounts as ConnectedAccountsV2,
+	ConnectedDids as ConnectedDidsV2, ConnectionRecordOf, DidIdentifierOf, MigrationOngoing, Pallet,
+};
 
 use frame_support::{
 	storage_alias,
 	traits::{Get, GetStorageVersion, OnRuntimeUpgrade},
 	Blake2_128Concat,
 };
-use sp_std::{marker::PhantomData, vec};
-
-#[cfg(feature = "try-runtime")]
-use frame_support::traits::OnRuntimeUpgradeHelpersExt;
-#[cfg(feature = "try-runtime")]
-use sp_runtime::SaturatedConversion;
+use sp_std::{marker::PhantomData, vec::Vec};
 
+/// Keytype changed from `AccountId` to `LinkableAccountId` changed in V3
 #[storage_alias]
 type ConnectedDids<T: Config> = StorageMap<Pallet<T>, Blake2_128Concat, AccountIdOf<T>, ConnectionRecordOf<T>>;
+/// Second keytype changed from `AccountId` to `LinkableAccountId` changed in V3
 #[storage_alias]
 type ConnectedAccounts<T: Config> =
 	StorageDoubleMap<Pallet<T>, Blake2_128Concat, DidIdentifierOf<T>, Blake2_128Concat, AccountIdOf<T>, ()>;
 
+/// Migrate the `ConnectedDids` and `ConnectedAccounts` key types for a given
+/// `AccountId`.
+pub(crate) fn do_migrate_account_id<T: Config>(
+	account_id: AccountIdOf<T>,
+	linkable_account: LinkableAccountId,
+) -> Option<DidIdentifierOf<T>> {
+	ConnectedDids::<T>::take(&account_id).map(|did_record| {
+		ConnectedDidsV2::<T>::insert(&linkable_account, did_record.clone());
+		if let Some(v) = ConnectedAccounts::<T>::take(&did_record.did, &account_id) {
```

Comment:
I think to be 100% safe, we should return an error (hence rollback to storage updates) if `ConnectedAccounts::<T>::take` returns `None`, as this should never happen if we are inside here and could indicate some other issues going on. Not failing means that we write `ConnectedDidsV2` but we don't write `ConnectedAccountsV2`, which are always suppoed to be in sync with each other.